src/System.Diagnostics.DiagnosticSource/ref/System.Diagnostics.DiagnosticSourceActivity.cs

```diff
@@ -9,18 +9,34 @@ namespace System.Diagnostics {
   public partial class Activity {
     public Activity(string operationName) {}      
     public string OperationName { get { throw null; } }
+
+    [System.ObsoleteAttribute]
     public string Id {get { throw null; } private set {} }
+
     public DateTime StartTimeUtc {get { throw null; } private set {} }
     public Activity Parent {get { throw null; } private set {} }
-    public string ParentId {get { throw null; } private set {} }
-    public string RootId {get { throw null; } private set {} }    
+  
+    public string ParentId {get { throw null; } set {} } // TraceId, public setter for binary (e.g. AMQP) or non-w3c custom propagation (e.g.Zipkin)
+
+    public string RootId {get { throw null; } set {} } // TraceId, public setter for binary (e.g. AMQP) or non-w3c custom propagation (e.g.Zipkin)
+
+    public string SpanId { get { throw null; } private set { } }
+
+    public byte TraceFlags {get { throw null; } set {} }
+
+    public string Tracestate {get { throw null; } set {} }
```

Comment:
Logically TraceState is baggage, however it serializes into its own header (so it is not like baggage that way), so it is definitely more convenient that it have its own property, so this design seems to be the best we can do.